src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/config/TelemetryService.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.config
+
+import com.microsoft.applicationinsights.TelemetryClient
+import mu.KotlinLogging
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.Goal
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Service class exposing methods to log telemetry events to ApplicationInsights.
+ */
+@Service
+class TelemetryService(
+  /*
+  The TelemetryClient is instantiated and injected at runtime by ApplicationInsights java agent via AppInsightsConfigManager
+  in the dps-gradle-spring-boot-plugin (https://github.com/ministryofjustice/dps-gradle-spring-boot/blob/spring-boot-3/src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/AppInsightsConfigManager.kt#L35)
+  where the jar is available at runtime courtesy of the Dockerfile. This means that in reality there will only be a TelemetryClient
+  bean in a deployed environment (ie. a deployed docker container).
+  In non-docker runtimes (eg: running in the IDE, or running the integration tests), this bean will not exist; hence the
+  nullable field and IDE warning.
+   */
+  private val telemetryClient: TelemetryClient?,
+) {
```

Comment:
Other services create a TelemetryClient bean to address this problem (eg: https://github.com/ministryofjustice/prison-api/blob/57bd1a2094efc613c0f4558f4c443ef9f2cd1338/src/main/java/uk/gov/justice/hmpps/prison/web/config/ApplicationInsightsConfiguration.kt) but 
* I don't think their documentation explains it well enough
* We don't need a telemetry client when running locally or integration tests (and we probably dont want one - dont want to be sending loads of events as part of our integration tests etc!)
* We don't like the idea of having production code that is not used in production

Even though my kdoc is a little wordy, I hope it explains it better than providing a dummy bean like the other services do.